middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -1578,6 +1482,244 @@ let close_let_rec acc env ~function_declarations
       named ~body
     |> Expr_with_acc.create_let
 
+let wrap_partial_application acc env apply_continuation (apply : IR.apply)
+    approx args missing_args ~arity ~num_trailing_local_params
+    ~contains_no_escaping_local_allocs =
+  (* In case of partial application, creates a wrapping function from scratch to
+     allow inlining and lifting *)
+  let wrapper_id = Ident.create_local ("partial_" ^ Ident.name apply.func) in
+  let closure_id =
+    Closure_id.wrap
+      (Compilation_unit.get_current_exn ())
+      (Variable.create_with_same_name_as_ident wrapper_id)
+  in
+  let args_arity = List.length args in
+  let params =
+    List.mapi
+      (fun n _ ->
+        ( Ident.create_local ("param" ^ string_of_int (args_arity + n)),
+          Lambda.Pgenval ))
+      missing_args
+  in
+  let return_continuation = Continuation.create ~sort:Return () in
+  let exn_continuation =
+    IR.{ exn_handler = Continuation.create (); extra_args = [] }
+  in
+  let all_args = args @ List.map (fun (a, _) -> IR.Var a) params in
+  let fbody acc env =
+    close_exact_or_unknown_apply acc env
+      { apply with
+        kind = Function;
+        args = all_args;
+        continuation = return_continuation;
+        exn_continuation
+      }
+      (Some approx)
+  in
+  let attr =
+    Lambda.
+      { inline = Default_inline;
+        specialise = Default_specialise;
+        local = Default_local;
+        is_a_functor = false;
+        stub = true
```

Comment:
The `stub` flag tends to be ignored outside of Simplify, but I think we should make sure that we don't generate stub functions that are not simple wrappers. In particular, in this case the full application in the wrapper function could get inlined by `close_exact_or_unknown_apply`, so it might be arbitrarily big in theory, and although in practice only small functions get inlined it would still not be a stub anymore.
So I would suggest either setting the `stub` attribute to false, or not going through `close_exact_or_unknown_apply` and generate directly the `Apply` node (which is what Closure does).